server/src-lib/Hasura/GraphQL/Validate/Types.hs

```diff
@@ -711,6 +713,7 @@ data AnnInpVal
   { _aivType     :: !G.GType
   , _aivVariable :: !(Maybe G.Variable)
```

Comment:
Let's change `G.Variable` to `VariableInfo` where `VariableInfo` represents `(G.Variable, Maybe AnnGValue)`.

Reply:
Currently, we seem to be adding this value [here](https://github.com/hasura/graphql-engine/pull/4620/files?file-filters%5B%5D=.hs&file-filters%5B%5D=.md&file-filters%5B%5D=.py&w=1#diff-10bf04471cd1721031b819698494806fL325), after this change we don't need to do that anymore, `validateVariables` itself can build this `VariableInfo` and add it to the `AnnInpVal`.

Reply:
Your link doesn't work for me, unfortunately, so I am not sure what you are suggesting.